src/time.rs

```diff
@@ -0,0 +1,117 @@
+use std::{
+    ops::Deref,
+    time::{Duration, Instant},
+};
+
+use bevy::prelude::*;
+
+use crate::{
+    AdvanceWorld, AdvanceWorldSet, CloneStrategy, ResourceSnapshotPlugin, RollbackFrameCount,
+    DEFAULT_FPS,
+};
+
+/// [`Resource`] describing the rate at which the [`AdvanceWorld`] will run.
+///
+/// Updating this value after the session has started may produce
+/// unintended results and should be avoided.
+#[derive(Resource, Clone, Copy, Debug, Hash)]
+pub struct RollbackFrameRate(pub usize);
```

Comment:
```suggestion
/// [`Resource`] describing the rate at which the [`AdvanceWorld`] will run.
#[derive(Resource, Clone, Copy, Debug, Hash, Deref)]
pub struct RollbackFrameRate(usize);
```

Maybe we should just make it impossible to modify or construct outside the crate?